### ISSUE TITLE
Reboot always tries to reboot to system after wiping /system

### ISSUE THREAD

**Issue by aykevl** [TeamWin/Team-Win-Recovery-Project#909]
TL;DR: when no `/system` is installed, TWRP always tries to reboot to system no matter which action is chosen in the Reboot screen.

After wiping the `/system` image, a prompt appears on each reboot action ("No OS Installed! Are you sure you wish to reboot?"). When swiping to reboot, the system hangs.
I think this is a bug in TWRP, as the LineageOS recovery doesn't have this issue and manually issuing `setprop sys.powerctl reboot,recovery` properly reboots the recovery.

Furthermore, I have figured out (with some strategically placed `sleep()` calls) that after the prompt, TWRP actually tries to reboot to system, not to recovery (or download, or poweroff). For example, when placing a `sleep(5)` in `twrp-functions.cpp`, under `case rb_system:` ([line 535](https://github.com/TeamWin/Team-Win-Recovery-Project/blob/android-6.0/twrp-functions.cpp#L535)), TWRP hangs for 5 seconds before rebooting. This indicates that TWRP tries to reboot to system and not to recovery. And the reboot obviously fails as there is no system installed.

I have tested this on a kminilte (G800F, Samsung Galaxy S5 mini). This device is not officially supported, but I'm trying to get it in a shape it *can* be officially supported.
